File: 70_Climbing_Stairs.py
class Solution:
    def climbStairs(self, n: int) -> int:
        oneStepBefore = 1
        twoStepsBefore = 0
        
        for _ in range(n-1):
            tmp = oneStepBefore
            oneStepBefore = oneStepBefore + twoStepsBefore
            twoStepsBefore = tmp
        
        return oneStepBefore + twoStepsBefore
        
        # recursive approach (time-complexity O(2^n)?)
        # if n == 0:
        #     return 0
        # elif n == 1:
        #     return 1
        # else:
        #     return self.climbStairs(n-1) + self.climbStairs(n-2)
        

File: swift/70_climbing_stairs.swift
class Solution {
    /// Returns the number of distinct ways to climb to the top of a staircase with n steps.
    func climbStairs(_ n: Int) -> Int {
        var oneStepBefore: Int = 1
        var twoStepBefore: Int = 0
        
        for _ in 0..<n-1 {
            var tmp = oneStepBefore
            oneStepBefore = oneStepBefore + twoStepBefore
            twoStepBefore = tmp
        }
        
        return oneStepBefore + twoStepBefore
    }
}

